tools/test/test_saas_promotion_state.py

```diff
@@ -0,0 +1,78 @@
+from collections.abc import (
+    Callable,
+    Iterable,
+    Mapping,
+)
+from unittest.mock import (
+    create_autospec,
+)
+
+from reconcile.typed_queries.saas_files import SaasFile
+from reconcile.utils.promotion_state import PromotionState
+from tools.saas_promotion_state.saas_promotion_state import (
+    SaasPromotionState,
+)
+
+
+def test_saas_promotion_state(
+    saas_files_builder: Callable[[Iterable[Mapping]], list[SaasFile]],
+) -> None:
+    saas_files = saas_files_builder([
+        {
+            "path": "/saas1.yml",
+            "name": "saas_1",
+            "resourceTemplates": [
+                {
+                    "name": "template_1",
+                    "url": "repo1/url",
+                    "targets": [
+                        {
+                            "ref": "main",
+                            "namespace": {"path": "/namespace1.yml"},
+                            "promotion": {
+                                "publish": ["channel-a"],
+                            },
+                        }
+                    ],
+                }
+            ],
+        },
+        {
+            "path": "/saas2.yml",
+            "name": "saas_2",
+            "resourceTemplates": [
+                {
+                    "name": "template_2",
+                    "url": "repo2/url",
+                    "targets": [
+                        {
+                            "ref": "main",
+                            "namespace": {"path": "/namespace2.yml"},
+                            "promotion": {
+                                "publish": ["channel-b"],
+                                "subscribe": ["channel-a"],
+                            },
+                        },
+                        {
+                            "ref": "main",
+                            "namespace": {"path": "/namespace3.yml"},
+                        },
+                    ],
+                }
+            ],
+        },
+    ])
+
+    promotion_state = create_autospec(spec=PromotionState)
+    promotion_state.get_promotion_data.return_value = ["data"]
+    saas_promotion_state = SaasPromotionState.create(
+        promotion_state=promotion_state, saas_files=saas_files
+    )
+    saas_promotion_state.get(channel="channel-a", sha="main")
```

Comment:
Do you want to verify expected output? May need to let `get` just return string and do actual print in cli method

Reply:
Good idea. Fixed :+1: